client/Commands/Prompts/AcceptDamagePrompt.ts

```diff
@@ -6,24 +6,28 @@ module ImprovedInitiative {
         private dequeueCallback: () => void;
 
         SetDequeueCallback = callback => this.dequeueCallback = callback;
+
         Resolve = (form: HTMLFormElement) => {
+            
             this.dequeueCallback();
         };
         AcceptFull: () => void;
         AcceptHalf: () => void;
 
-        constructor(combatantNames: string, damageAmount: number, suggester: string, resolve: (responses: { [id: string]: string }) => void) {
+        constructor(suggestedCombatants: CombatantViewModel [], damageAmount: number, suggester: string) {
+            const combatantNames = suggestedCombatants.map(c => c.Name()).join(', ');
             const displayType = (damageAmount < 0) ? "healing" : "damage";
             const displayNumber = (damageAmount < 0) ? -damageAmount : damageAmount;
             this.Prompt = `Accept ${displayNumber} ${displayType} to ${combatantNames} from ${suggester}?`;
 
             this.AcceptFull = () => {
-                resolve({ 'damage': `${damageAmount}` });
+                suggestedCombatants.forEach(c => c.ApplyDamage(damageAmount.toString()))
```

Comment:
As far as I can tell, this change prevents the player view from updating immediately when the change is approved.  Additionally, it doesn't show up in the event log like this.  That's why I made it go through that sort of convoluted callback.